python/src/aica_api/client.py

```diff
@@ -119,14 +179,21 @@ def core_version(self) -> Union[str, None]:
         """
         core_version = None
         try:
-            core_version = requests.get(f'{self._address}/version').json()
+            core_version = requests.get(self.__raw_endpoint('version')).json()
+        except requests.exceptions.InternalServerError:
+            # most likely an unsigned core
+            pass
```

Comment:
If the server returns 500 it does not raise an exception here, so these lines are not necessary (tested it locally, this path was never executed)

Reply:
It is with a non-signed backend as seen by @domire8 and I

Reply:
I have seen the error on the server side, not client